exodus_lambda/functions/verify_source.py

```diff
@@ -0,0 +1,56 @@
+import json
+import os
+import subprocess
+
+import boto3
+
+
+def lambda_handler(event, context):
+    """Verifies the commit delivered to the pipeline by GitHub has a
+    tag signed by a trusted entity.
+
+    Receives the commit ID from the pipeline via UserParameters.
+
+    This function must be manually deployed prior to pipeline creation.
+    """
+    # pylint: disable=unused-argument
+
+    job_id = event["CodePipeline.job"]["id"]
+    job_data = event["CodePipeline.job"]["data"]
+    job_params = json.loads(
+        job_data["actionConfiguration"]["configuration"]["UserParameters"]
+    )
+
+    pipeline_client = boto3.client("codepipeline")
+
+    try:
+        # Import whitelisted GPG keys
+        pub_keys = "".join(
+            pub_key for _, pub_key in job_params["whitelist"].items()
+        )
+        subprocess.run(["gpg", "--import", pub_keys], check=True)
```

Comment:
Hmm, sorry to ask but does it work?  If I understand right, pub_keys is going to be a string like 

```
-----BEGIN PGP PUBLIC KEY BLOCK-----
... [key1]
-----END PGP PUBLIC KEY BLOCK-----
-----BEGIN PGP PUBLIC KEY BLOCK-----
... [key2]
-----END PGP PUBLIC KEY BLOCK-----
```

I am surprised that gpg would accept this entire string as a command-line argument.  I would have expected you need to feed it to stdin, or save it to a file first and point gpg at the file, or similar.  The argument after --import is usually interpreted as a filename.

Reply:
I've confirmed that they needs to be stored in a file.